### DISCUSSION TITLE
Past year of activity chart.

### DISCUSSION THREAD

**Discussion by damquanghieu** [community/community#61594] [Discussions] [answered]
### Select Topic Area

Question

### Body

Can anyone tell me what data the "Past year of activity chart." is generated from?

![image](https://github.com/community/community/assets/65754078/4118a0ea-bbfe-4659-9043-b4163e7b0b10)

**Comment by nirav-gajera** [comment]
The "Past year of activity chart" in is generated from the user's or repository's activity over the past year. It shows a visual representation of the user's or repository's activity in terms of contributions and interactions on GitHub.

The chart includes various types of activities, such as:

1. Commits: The number of code commits made to repositories.
2. Pull Requests: The number of pull requests opened and merged.
3. Issues: The number of issues opened and closed.
4. Comments: The number of comments made on pull requests, issues, and commits.
5. Reviews: The number of pull request reviews given.

The chart provides a quick overview of the user's or repository's engagement and contribution to GitHub over the past year.

**Reply by damquanghieu** [reply]
Thanks for your help. Can you give me a document that describes this type of chart in detail. I tried aggregating my data but it's not nearly the same as the histogram.

**Reply by nirav-gajera** [reply]
-github document itself is good for understanding things in detail

https://docs.github.com/en/apps

https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/viewing-contributions-on-your-profile

**Reply by damquanghieu** [reply]
![image](https://github.com/community/community/assets/65754078/e7787a69-505b-4ca3-a772-1bdbba587118)
Thank you very much. I tried searching all in github. And the link you sent but can't find where the "Past year of activity" chart is explained. This chart is shown next to the archive

**Reply by damquanghieu** [reply]
I have a task to analyze this chart. I would like to be able to have a concrete example of how that graph was created?

**Reply by nirav-gajera** [reply]
https://stackoverflow.com/questions/61936264/understanding-the-git-graph
https://docs.github.com/en/repositories/viewing-activity-and-data-for-your-repository

**Reply by nirav-gajera** [reply]
https://docs.github.com/en/repositories/viewing-activity-and-data-for-your-repository/analyzing-changes-to-a-repositorys-content

**Reply by damquanghieu** [reply]
So do you think the "Past year of activity" chart represents commits in a year?

**Reply by nirav-gajera** [reply, marked as answer]
Yes, Past year of activity chart represents commits in a year. The chart shows the number of commits made to a repository each day over the past year. The darker squares represent the days with the most commits, and the lighter squares represent the days with fewer commits. The chart can be used to see how active a repository is and to identify days when there were a lot of commits.

**Reply by damquanghieu** [reply]
Thanks so much
